lib/index.ts

```diff
@@ -131,6 +130,44 @@ class LogService {
     }
   }
 
+  addLogger(
+    loggerScopeFunction: (config: InitialConfig) => LoggerPlugin,
+    pluginConfig: PluginConfig) {
+    if (typeof loggerScopeFunction !== 'function')
+      return;
+
+    const loggerOptions = loggerScopeFunction.call(this, this.globalConfig);
+
+    if (loggerOptions == null)
+      return;
+
+    const loggerKeys = Object.keys(loggerOptions);
+    const requiredKeys = ['error', 'warn', 'info', 'debug'];
+    const isValidObject = requiredKeys
+      .every((method) => loggerKeys.indexOf(method) !== -1)
+
+    if (!isValidObject)
+      throw new Error(
+        `Invalid Logger, missing one or more of required keys: [${requiredKeys.join(', ')}]`
```

Comment:
instead of saying "missing ... required keys", what you think about "missing one or more of required methods"?

Reply:
In this case I want to present which fields are missing... saying as you said it's too gerenic, it's easier when the dev knows what fields are missing

Reply:
> In this case I want to present which fields are missing... saying as you said it's too gerenic, it's easier when the dev knows what fields are missing

yha, he will know which are the missing methods, I was suggesting only replacing the word "keys" by "methods".

Reply:
It gave me another idea, instead of `[${requiredKeys.join(', ')}]` put only the keys missing, not ALL the required.